ha/plugin/hare/hare.py

```diff
@@ -0,0 +1,103 @@
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+"""
+ ****************************************************************************
+ Description:       Hare plugin to provide Hare utility and interfaces.
+ ****************************************************************************
+"""
+
+import json
+from cortx.utils.log import Log
+from ha import const
+
+class Hare:
+
+    @staticmethod
+    def get_fid(service_name: str, node_id: str, instance_id: int) -> str:
+        """
+        Get Fid from hare mapping file for hare services.
+
+        Args:
+            service_name ([str]): Service name.
+            node_id ([str]): Node name for fid instance.
+            instance_id ([int]): Instance id for service.
+
+        Returns:
+            str: Return fid for hare service.
+        """
+        try:
+            with open(const.HARE_FID_MAPPING_FILE) as fi:
+                fid_schema = json.load(fi)
+            fid: str = ""
+            if service_name == "s3service":
+                fid = Hare._get_s3server_fid(fid_schema, service_name, node_id, instance_id)
+            else:
+                fid = Hare._get_motr_fid(fid_schema, service_name, node_id, instance_id)
+            Log.debug(f"Map ({service_name}, {node_id}, {instance_id}) to {fid}")
+            return fid
+        except Exception as e:
+            Log.error(f"Failed to get fid for ({service_name}, {node_id}, \
+                {instance_id}). Error: {e}")
+            return None
+
+    @staticmethod
+    def _get_motr_fid(fid_schema: dict, service_name: str, node_id: str, instance_id: int) -> str:
+        """
+        Get Fid from hare mapping file for motr services.
+
+        Args:
+            fid_schema ([dict]): Schema for hare services fid.
+            service_name ([str]): Service name.
+            node_id ([str]): Node name for fid instance.
+            instance_id ([int]): Instance id for service.
+
+        Returns:
+            str: Return fid for motr
+        """
+        motr_mapping: dict = {}
+        count: int = 0
+        for service in fid_schema["services"]:
+            if service["name"] == service_name:
+                count += 1
+                motr_mapping[count] = service["checks"][0]["args"][2]
```

Comment:
We should get the list of Fids and sort those and then assign it to instance ids in increasing order.
The current logic doesn't have a problem in VM but could be problematic in multi-node when fid_mapping file is created on all the nodes and order is different. Same is applicable for s3server fids.

Reply:
sort() may not need 
as per comment from hare team
```
So for single node I think it should be fine because you will find everything on just one node but for multinode such file is created per node. Every node runs a consul, hax, motr and s3 services thus consul-server-conf.json is generated per node. So, a consul-server-conf.json on a node will have only that node's service ids and to get service ids for all the nodes you would have to read consul-server-conf.json from all the nodes, which is not very good.
```
Every node will have its own fid only
so logic is same as single node
Also we have node_id to filter fid or conf_store

Reply:
I still think that we should have control on sequence of fids. If it takes time, then open a github issue and this can be implemented later.

Reply:
Added sort logic